File: Projekt/src/main/java/de/neosit/minecraft/arduinointegration/Configuration.java
package de.neosit.minecraft.arduinointegration;

import java.util.Arrays;

import org.bukkit.configuration.file.FileConfiguration;

/**
 * Konfigurationsklasse.
 * Stellt Zugriff auf die Konfigurationsdatei sicher.
 *
 */
public class Configuration {
	public static final String PORT = "port";
	public static final String TIMEOUT = "timeout";
	public static final String DATA_RATE = "data_rate";
	public static final String PERIOD = "period";
	public static final String ARDUINO_PREFIX = "arduino_";
	
	private FileConfiguration config;

	public Configuration(FileConfiguration config) {
		this.config = config;
	}
	
	public void setDefaults() {
		config.addDefault(PORT, Arrays.asList("COM3"));
		config.addDefault(TIMEOUT, 1000);
		config.addDefault(DATA_RATE, 9600);
		config.addDefault(PERIOD, 4);
		config.options().copyDefaults(true);
	}
	
	public String getPort() {
		return (String) config.get(PORT);
	}
	
	public int getTimeout() {
		return (int) config.getInt(TIMEOUT);
	}
	
	public int getDataRate() {
		return (int) config.getInt(DATA_RATE);
	}
	
	/**
	 * Gibt den Befehl zurück, der zu einem bestimmten 
	 * Element in der Konfigurationsdatei hinterlegt ist.
	 * 
	 * @param arduinoValue Ist der Wert, der direkt vom Arduino gesendet wird.
	 * @return Gibt den Befehl zurück, der in der Konfiguration hinterlegt ist.
	 */
	public String getCommandForArduinoValue(String arduinoValue) {
		String key = ARDUINO_PREFIX + arduinoValue;
		return (String) config.getString(key);
	}
	
	public int getPeriod() {
		return (int) config.getInt(PERIOD);
	}
}

File: Projekt/src/main/java/de/neosit/minecraft/arduinointegration/Configuration.kt
package de.neosit.minecraft.arduinointegration

import java.util.Arrays

import org.bukkit.configuration.file.FileConfiguration

/**
 * Konfigurationsklasse.
 * Stellt Zugriff auf die Konfigurationsdatei sicher.
 *
 */
class Configuration(private var config: FileConfiguration) {

    val ports: Array<String>
        get() = config.getStringList(PORT).toTypedArray()

    val timeout: Int
        get() = config.getInt(TIMEOUT)

    val dataRate: Int
        get() = config.getInt(DATA_RATE)

    val period: Int
        get() = config.getInt(PERIOD)

    fun setDefaults() {
        config.addDefault(PORT, Arrays.asList("/dev/tty.usbmodem", "/dev/usbdev", "/dev/tty", "/dev/serial", "COM3"))
        config.addDefault(TIMEOUT, 1000)
        config.addDefault(DATA_RATE, 9600)
        config.addDefault(PERIOD, 4)
        config.options().copyDefaults(true)
    }

    fun reload(config: FileConfiguration) {
        this.config = config
    }

    /**
     * Gibt den Befehl zurück, der zu einem bestimmten
     * Element in der Konfigurationsdatei hinterlegt ist.
     *
     * @param arduinoValue Ist der Wert, der direkt vom Arduino gesendet wird.
     * @return Gibt den Befehl zurück, der in der Konfiguration hinterlegt ist.
     */
    fun getCommandForArduinoValue(arduinoValue: String): String {
        val key = ARDUINO_PREFIX + arduinoValue
        return config.getString(key) as String
    }
}

const val PORT = "port"
const val TIMEOUT = "timeout"
const val DATA_RATE = "data_rate"
const val PERIOD = "period"
const val ARDUINO_PREFIX = "arduino_"
